### PR TITLE
support redis 4.0.0

### PR DESCRIPTION
**Pull request by ota42y**
delete redis version lock from redis-namespace 1.6
So we can remove redis version lock and use redis gem 4.0.0 :)

### CODE REVIEW

File: redis-objects.gemspec [hunk 1]
```diff
@@ -19,7 +19,7 @@ Gem::Specification.new do |spec|
   spec.require_paths = ["lib"]
 
   # Only fix this one version or else tests break
-  spec.add_dependency "redis", "~> 3.3"
+  spec.add_dependency "redis"
```
Comment (D-system) [inline redis-objects.gemspec]: Better to always specify a version. If the 4.x versions works then could be nice to write it this way:
`spec.add_dependency "redis", "~> 3.3", "< 5.0"`

Comment (ota42y) [reply redis-objects.gemspec]: I did fix! thanks!

**Comment by ota42y** [comment]
@nateware 

Hi!
Do you have problem?

File: Gemfile [hunk 2]
```diff
@@ -2,3 +2,7 @@ source 'https://rubygems.org'
 
 # Specify your gem's dependencies in redis-objects.gemspec
 gemspec
+
+group :development, :test do
+  gem 'redis-namespace', git: 'https://github.com/resque/redis-namespace'
```
Comment (MT-Damax) [inline Gemfile]: @ota42y why did you move `redis-namespace` from the `gemspec` file to the `Gemfile`?

File: redis-objects.gemspec [hunk 3]
```diff
@@ -28,7 +28,6 @@ Gem::Specification.new do |spec|
   spec.add_development_dependency "connection_pool"
 
   # Compatibility testing
-  spec.add_development_dependency "redis-namespace"
```
Comment (D-system) [inline redis-objects.gemspec]: Let's use the gem from rubygems.org instead of git
The 1.6 version is released and can use `redis` 4 (thanks to you @ota42y さん https://github.com/resque/redis-namespace/pull/136)

```
spec.add_development_dependency "redis-namespace", "~> 1.6"
```

Comment (ota42y) [reply redis-objects.gemspec]: I fixed.
But I don't set `"~> 1.6"` because this gem work well 1.5.
So the user using 1.5 can upgrade redis-object without upgrading redis-namespace :)

File: redis-objects.gemspec [hunk 4]
```diff
@@ -19,7 +19,7 @@ Gem::Specification.new do |spec|
   spec.require_paths = ["lib"]
 
   # Only fix this one version or else tests break
-  spec.add_dependency "redis", "~> 3.3"
+  spec.add_dependency "redis", "~> 3.3", "< 5.0"
```
Comment (D-system) [inline redis-objects.gemspec]: It should be `> 3.3` (no `~`)

Comment (ota42y) [reply redis-objects.gemspec]: oh... 😢
